xscen/utils.py

```diff
@@ -733,7 +733,12 @@ def _search(a, b):
                     del ds.attrs[a_key]
 
         # generate a new id
-        ds.attrs["cat:id"] = generate_id(ds).iloc[0]
+        try:
+            ds.attrs["cat:id"] = generate_id(ds).iloc[0]
+        except IndexError as err:
+            logger.warning(f"Unable to generate a new id for the dataset. Got {err}.")
```

Comment:
A use case here would be if someone calls `xs.ensemble_stats(common_attrs_only)` with a list of paths or Datasets. `common_attrs_only` is useful in that instance to clean up the results, but it will crash since the datasets have not been through `intake-esm`.